doc/semiboolmat.xml

```diff
@@ -0,0 +1,150 @@
+#############################################################################
+##
+#W  semiboolmat.xml
+#Y  Copyright (C) 2017                                      Wilf A. Wilson
+##
+##  Licensing information can be found in the README file of this package.
+##
+#############################################################################
+##
+
+<#GAPDoc Label="RegularBooleanMatMonoid">
+<ManSection>
+  <Oper Name="RegularBooleanMatMonoid" Arg="d"/>
+  <Returns>A monoid of boolean matrices.</Returns>
+  <Description>
+    If <A>d</A> is a positive integer, then
+    <C>RegularBinaryRelationSemigroup</C> returns the monoid generated by the
+    regular <A>d</A> by <A>d</A> boolean matrices. Note that this monoid is
+    <E>not</E> regular in general.  <C>RegularBooleanMatMonoid(<A>d</A>)</C> is
+    generated by the four boolean matrices <C>A, B, C, D</C>, whose <K>true</K>
+    entries are:
+    <List>
+      <Item>
+        <C>A[i][i + 1]</C> and <C>A[n][1]</C>, for
+        <M>i \in \{1, \ldots, n - 1\}</M>;
+      </Item>
+      <Item>
+        <C>B[1][2]</C>, <C>B[2][1]</C>, and <C>B[i][i]</C> for <M>i \in \{3,
+          \ldots, n\}</M>;
+      </Item>
+      <Item>
+        <C>C[1][2]</C> and <C>C[i][i]</C>, for <M>i \in \{2, \ldots, n -
+          1\}</M>; and
+      </Item>
+      <Item>
+        <C>D[1][2]</C>, <C>D[i][i]</C>, for <M>i \in \{2, \ldots, n\}</M>,
+        and <C>D[n][1]</C>.
+      </Item>
+    </List>
+    This monoid has nearly <C>2 ^ (n ^ 2)</C> elements. 
+  </Description>
+</ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="FullBooleanMatMonoid">
+   <ManSection>
+    <Oper Name="FullBooleanMatMonoid" Arg="d"/>
+    <Returns>A monoid of boolean matrices.</Returns>
+    <Description>
+      If <A>d</A> is a positive integer, then this operation returns full
+      boolean matrix monoid, which is the monoid consisting of all <A>d</A> by
+      <A>d</A> boolean matrices. This monoid consists of <C>2 ^ (<A>n</A> ^
+        2)</C> elements.
+
```

Comment:
This only works when `d <= 5`, I think, and the generators are a minimum in number, and stored in a file (the latter is probably not very important.